bretm9/my-tier-ultimate
# Testcafe setup

### This PRs additions:
* TestCafe dependency
* Initial testing for Header as a starting point for TestCafe implementation
 
### Reviewer starting point: 
`src/Header/Header.e2e.tsx`

### How has this been tested:  
* Manually verified that all routes still function as expected
* `npm run test` still runs only Jest tests and all tests are still passing
* `testcafe chrome src/Header/Header.e2e.tsx` runs our new simple TestCafe end-to-end tests and they are passing.
 
### How to manually test:  
* run `npm install -g testcafe` to ensure TestCafe is installed globally on your local environment
* run `npm i` to ensure that TestCafe is installed in the project
* run `testcafe chrome src/Header/Header.e2e.tsx` and observe that the chrome browser opens and two simple end-to end `Header` tests pass.
 
### background context:  
Line 4 of `src/Header/Header.e2e.tsx` declares the type for `test` within the scope of the file. This is necessary because otherwise `test`'s type defaults to the type declared by `@types/jest`. This declaration ensures that `test` is typed for TestCafe and not Jest. This line will be necessary in every end-to-end test file in this repo unless our dependencies versioning is updated
 
### Relevant tickets: 
Closes #113 
 
### Screenshots:
N/A
 
### Questions: 
N/A
